graphql/resolvers.go

```diff
@@ -357,6 +359,52 @@ func (r *queryResolver) TaskFiles(ctx context.Context, taskID string) ([]*Groupe
 	return groupedFilesList, nil
 }
 
+func (r *mutationResolver) SetTaskPriority(ctx context.Context, taskID string, priority int) (*restModel.APITask, error) {
+	t, err := r.sc.FindTaskById(taskID)
+	if err != nil {
+		return nil, ResourceNotFound.Send(ctx, err.Error())
+	}
+	if t == nil {
+		return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("cannot find task with id %s", taskID))
+	}
+	authUser := gimlet.GetUser(ctx)
+	if authUser == nil {
+		return nil, Forbidden.Send(ctx, "No user attached to request")
+	}
+	if priority > evergreen.MaxTaskPriority {
+		requiredPermission := gimlet.PermissionOpts{
+			Resource:      t.Project,
+			ResourceType:  "project",
+			Permission:    evergreen.PermissionTasks,
+			RequiredLevel: evergreen.TasksAdmin.Value,
+		}
+		isTaskAdmin := authUser.HasPermission(requiredPermission)
+		if !isTaskAdmin {
+			return nil, Forbidden.Send(ctx, fmt.Sprintf("Insufficient access to set priority %v, can only set priority less than or equal to %v", priority, evergreen.MaxTaskPriority))
+		}
+	}
+	if err = t.SetPriority(int64(priority), authUser.Username()); err != nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error setting task priority %v: %v", taskID, err.Error()))
+	}
+	t, err = r.sc.FindTaskById(taskID)
```

Comment:
Since SetPriority also changes the task in-memory, there's no need to do this query again